mixer/pkg/runtime2/routing/monitoring.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package routing
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	meshFunction = "meshFunction"
+	handlerName  = "handler"
+	adapterName  = "adapter"
+)
+
+var (
+	durationBuckets    = []float64{.0001, .00025, .0005, .001, .0025, .005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10}
+	dispatchLabelNames = []string{meshFunction, handlerName, adapterName}
```

Comment:
should add `error` boolean to dimensions, and remove `failedDispatchCount`.
That way be can slide both `dispatchCount` and ` dispatchDuration` metric by failures if needed.

Also we have been adding (should add) error booleans for all these metrics.